server/server.go

```diff
@@ -821,7 +832,9 @@ func (s *StanServer) createNatsClientConn(name string, sOpts *Options, nOpts *se
 		ncOpts.ReconnectBufSize = 128
 	}
 
-	Tracef(" NATS conn opts: %v", ncOpts)
+	if s.trace {
```

Comment:
If you want to check what the server's `debug` and `trace` flags are, you would find this in the `StanServer` structure. I think that's enough:
```
	// Use these flags for Debug/Trace in places where speed matters.
	// Normally, Debugf and Tracef will check an atomic variable to
	// figure out if the statement should be logged, however, the
	// cost of calling Debugf/Tracef is still significant since there
	// may be memory allocations to format the string passed to these
	// calls. So in those situations, use these flags to surround the
	// calls to Debugf/Tracef.
```
I will remove the `atomic` reference. In this PR, I also surrounded the Debug print statements about subjects with `if s.debug`, but that was unnecessary. As the comment suggest, you would want to do that only when speed matters (that is, in the fast path). I am going to look at the place where I use the server's debug/trace and remove it in place where it is not really needed.